### DISCUSSION TITLE
[New Feature] Track and export information of the "strength" of players during the game

### DISCUSSION THREAD

**Discussion by Rampastring** [Vinifera-Developers/Vinifera#1382] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

C&C3 has this cool feature of tracking statistics that provide a nice overview of a multiplayer match in the score screen.

![Unit count over time](https://cdn.discordapp.com/attachments/878792778203672607/969284520165646396/unknown.png)

![Structure count over time](https://cdn.discordapp.com/attachments/878792778203672607/969284561995436083/unknown.png)

![Total resources harvested over time](https://cdn.discordapp.com/attachments/878792778203672607/969284603921694781/unknown.png)

![Various other statistics](https://cdn.discordapp.com/attachments/878792778203672607/969284647441825802/unknown.png)

Of these, the most interesting one is the unit count over time.

However I think even that is a bit unoptimal. If player A has 100 Light infantry and player B has 50 Disruptors, player A would appear to be twice as strong in the chart, while actually player B's army would be much more powerful. It might be better if the total cost of the player's army over time was tracked instead.

### Proposed Code (optional)

I can think of the having a separate tracker class for tracking the total cost of a player's objects. Whenever the player gains new objects, their value should be added to the total, and whenever the player loses an object, the object's value should be deducted from the total. Each instance when this happens (the total value changes), the new total as well as the frame # when this change happened should be saved.

Basically this kind of a tracker could consist of a list (vector) of structs like this

```cpp
struct TotalObjectValueChangeStruct
{
    long TotalValue;
    long Frame;
}
```

and this list would then be exported at the end of the game for all players. (See #780 )

Another option is that instead of tracking and exporting each change of the total value, there'd be a "snapshot" value saved every X frames if the value is different from what it was in the previous snapshot. After all, knowing the exact frames when the value has been changed is a bit overkill for stat-displaying purposes. Instead the game could check every X frames (like 300 frames for 5 seconds at the typical 60 FPS online gamespeed) whether the total value has been changed and if so, then add a new entry to the list.
